derive/src/derive_enum.rs

```diff
@@ -168,7 +177,7 @@ impl DeriveEnum {
                     // no fixed values, implement a range
                     variant_inner.push_parsed(format!(
                         "bincode::error::AllowedEnumVariants::Range {{ min: 0, max: {} }}",
-                        self.variants.len() - 1
+                        self.variants.len().saturating_sub(1)
```

Comment:
Wouldn't this make more sense as an error dedicated specifically to "Tried to deserialize an uninhabited enum?"

Reply:
I agree, I also made is to that it doesn't read data from the reader, because this isn't needed

```rs
impl bincode::Decode for EmptyEnum {
    fn decode<D: bincode::de::Decoder>(
        mut decoder: D,
    ) -> core::result::Result<Self, bincode::error::DecodeError> {
        core::result::Result::Err(bincode::error::DecodeError::EmptyEnum {
            type_name: core::any::type_name::<Self>(),
        })
    }
}
```

Added a test for this as well